pkg/ibmvpc/vpc.go

```diff
@@ -286,38 +287,102 @@ func (nl *NaclLayer) RulesInConnectivity(src, dst vpcmodel.Node,
-			Filter: filterIndex,
-			Rules:  *rules,
-		}
-		*resRulesInFilter = append(*resRulesInFilter, rulesInNacl)
+func appendToRulesInFilter(tableRelevant bool, resRulesInFilter *[]vpcmodel.RulesInFilter,
+	rules *[]int, filterIndex int, isAllow bool) {
+	if !tableRelevant {
+		return
+	}
+
+	var rType vpcmodel.RulesType
+	switch {
+	case len(*rules) == 0:
+		rType = vpcmodel.NoRules
+	case len(*rules) == 1 && (*rules)[0] == -1:
+		rType = vpcmodel.OnlyDummyRule
+	case isAllow:
+		rType = vpcmodel.OnlyAllow
+	default: // more than 0 deny rules
+		rType = vpcmodel.OnlyDeny
+	}
+	rulesInNacl := vpcmodel.RulesInFilter{
+		Filter: filterIndex,
+		Rules:  *rules,
+		RType:  rType,
 	}
+	*resRulesInFilter = append(*resRulesInFilter, rulesInNacl)
 }
 
-func (nl *NaclLayer) StringRulesOfFilter(listRulesInFilter []vpcmodel.RulesInFilter) string {
+const networkACLStr = "network ACL "
+const securityGroupStr = "security group "
+
+func (nl *NaclLayer) StringDetailsRulesOfFilter(listRulesInFilter []vpcmodel.RulesInFilter) string {
 	strListRulesInFilter := ""
 	for _, rulesInFilter := range listRulesInFilter {
 		nacl := nl.naclList[rulesInFilter.Filter]
-		strListRulesThisNacl := nacl.analyzer.StringRules(rulesInFilter.Rules)
-		if strListRulesThisNacl != "" {
-			strListRulesInFilter += rulesOfFilterHeader("network ACL "+nacl.Name()) + strListRulesThisNacl
+		header := getHeaderRulesType(networkACLStr+nacl.Name(), rulesInFilter.RType) +
+			nacl.analyzer.StringRules(rulesInFilter.Rules)
+		if header == "" {
+			continue // only dummy rule - nacl not needed between two vsis of the same subnet
 		}
+		strListRulesInFilter += getHeaderRulesType(networkACLStr+nacl.Name(), rulesInFilter.RType) +
+			nacl.analyzer.StringRules(rulesInFilter.Rules)
 	}
 	return strListRulesInFilter
 }
 
+func (nl *NaclLayer) StringFilterEffect(listRulesInFilter []vpcmodel.RulesInFilter) string {
+	filtersEffectList := []string{}
+	for _, rulesInFilter := range listRulesInFilter {
+		nacl := nl.naclList[rulesInFilter.Filter]
+		header := getSummaryFilterEffect(networkACLStr+nacl.Name(), rulesInFilter.RType)
+		if header == "" {
+			continue // only dummy rule - nacl not needed between two vsis of the same subnet
+		}
+		filtersEffectList = append(filtersEffectList, header)
+	}
+	return strings.Join(filtersEffectList, semicolonSeparator)
+}
+
+func getHeaderRulesType(filter string, rType vpcmodel.RulesType) string {
+	switch rType {
+	case vpcmodel.NoRules:
+		return filter + " blocks connection since there are no relevant allow rules\n"
+	case vpcmodel.OnlyDeny:
+		return filter + " blocks connection with the following deny rules:\n"
+	case vpcmodel.BothAllowDeny:
+		return filter + " allows connection with the following allow and deny rules\n"
+	case vpcmodel.OnlyAllow:
+		return filter + " allows connection with the following allow rules\n"
+	default:
+		return "" // OnlyDummyRule
+	}
+}
+
+func getSummaryFilterEffect(filter string, rType vpcmodel.RulesType) string {
```

Comment:
please keep the order of the file 
both getHeaderRulesType() and getSummaryFilterEffect() are not spcific for  NaclLayer

Reply:
ok